docs/content/concepts/automation/declarative-automation.mdx

```diff
@@ -0,0 +1,182 @@
+---
+title: "Declarative Automation | Dagster Docs"
+description: "Dagster can automatically materialize assets when criteria are met, enabling a declarative approach to asset materialization."
+---
+
+# Declarative Automation
+
+<Note>
+  This feature is currently <strong>experimental</strong>.
+</Note>
+
+Dagster can automatically materialize assets when criteria are met, enabling a declarative approach to asset materialization. Instead of defining explicit workflows to materialize assets, you describe the conditions under which they should be materialized and let the system kick off runs in response.
+
+Declarative Automation includes pre-built conditions to handle common use cases, such as on a periodic schedule or whenever an upstream dependency updates, but conditions can also be customized.
+
+---
+
+## Benefits
+
+Using Declarative Automation helps you:
+
+- Define the precise conditions under which an asset should be materialized
+- Define all logic related to an asset in a single place
+- Avoid thinking about specific workflow boundaries
+
+---
+
+## Prerequisites
+
+Before continuing, you should be familiar with:
+
+- [Asset definitions](/concepts/assets/software-defined-assets)
+- [Sensor definitions](/concepts/partitions-schedules-sensors/sensors)
+- [Code locations](/concepts/code-locations)
+
+---
+
+## How it works
+
+Declarative Automation is an automation method that kicks off runs when criteria are met and contains two main components:
+
+- **An automation condition (<PyObject object="AutomationCondition" />**), which represents when an individual asset should be executed.
+- **A sensor** which evaluates each <PyObject object="AutomationCondition" /> and launches runs in response to their status.
```

Comment:
I think this would be a good place to link to the API docs for `AutomationConditionSensorDefinition` so that we directly link to both API docs here (rather than having two links to AutomationCondition).

Maybe "**A sensor (<PyObject object="AutomationConditionSensorDefinition">)**, which evaluates each condition and launches runs in response to their status.